src/BuildOpt.h

```diff
@@ -105,833 +65,218 @@
-    #define RADIOLIB_CB_ARGS_DETACH_INTERRUPT           (void, detachInterrupt, uint8_t interruptNum)
-    #define RADIOLIB_CB_ARGS_YIELD                      (void, yield, void)
-    #define RADIOLIB_CB_ARGS_DELAY                      (void, delay, unsigned long ms)
-    #define RADIOLIB_CB_ARGS_DELAY_MICROSECONDS         (void, delayMicroseconds, unsigned int us)
-    #define RADIOLIB_CB_ARGS_MILLIS                     (unsigned long, millis, void)
-    #define RADIOLIB_CB_ARGS_MICROS                     (unsigned long, micros, void)
-    #define RADIOLIB_CB_ARGS_PULSE_IN                   (unsigned long, pulseIn, uint8_t pin, uint8_t state, unsigned long timeout)
-    #define RADIOLIB_CB_ARGS_SPI_BEGIN                  (void, SPIbegin, void)
-    #define RADIOLIB_CB_ARGS_SPI_BEGIN_TRANSACTION      (void, SPIbeginTransaction, void)
-    #define RADIOLIB_CB_ARGS_SPI_TRANSFER               (uint8_t, SPItransfer, uint8_t b)
-    #define RADIOLIB_CB_ARGS_SPI_END_TRANSACTION        (void, SPIendTransaction, void)
-    #define RADIOLIB_CB_ARGS_SPI_END                    (void, SPIend, void)
+  // ... and for the grand finale, we have millis() and micros() DEFINED AS MACROS!
+  #if defined(millis)
+  #undef millis
+  inline unsigned long millis() { return((unsigned long)(STCV / 1000)); };
+  #endif
 
+  #if defined(micros)
+  #undef micros
+  inline unsigned long micros() { return((unsigned long)(STCV)); };
   #endif
+#elif defined(TEENSYDUINO)
+  // Teensy
+  #define RADIOLIB_PLATFORM                           "Teensy"
+#else
+  // other Arduino platforms not covered by the above list - this may or may not work
+  #define RADIOLIB_PLATFORM                           "Unknown Arduino"
+  #define RADIOLIB_UNKNOWN_PLATFORM
 #endif
 
+  #if !defined(RADIOLIB_NC)
+  #define RADIOLIB_NC                                 (0xFF)
+  #endif
+  #if !defined(RADIOLIB_DEFAULT_SPI)
+  #define RADIOLIB_DEFAULT_SPI                        SPI
+  #endif
+  #if !defined(RADIOLIB_DEFAULT_SPI_SETTINGS)
+  #define RADIOLIB_DEFAULT_SPI_SETTINGS               SPISettings(2000000, MSBFIRST, SPI_MODE0)
+  #endif
+  #if !defined(RADIOLIB_NONVOLATILE)
+  #define RADIOLIB_NONVOLATILE                        PROGMEM
+  #endif
+  #if !defined(RADIOLIB_NONVOLATILE_READ_BYTE)
+  #define RADIOLIB_NONVOLATILE_READ_BYTE(addr)        pgm_read_byte(addr)
+  #endif
+  #if !defined(RADIOLIB_NONVOLATILE_READ_DWORD)
+  #define RADIOLIB_NONVOLATILE_READ_DWORD(addr)       pgm_read_dword(addr)
+  #endif
+  #if !defined(RADIOLIB_TYPE_ALIAS)
+  #define RADIOLIB_TYPE_ALIAS(type, alias)            using alias = type;
+  #endif
+
+  #if !defined(RADIOLIB_ARDUINOHAL_PIN_MODE_CAST)
+  #define RADIOLIB_ARDUINOHAL_PIN_MODE_CAST
+  #endif
+  #if !defined(RADIOLIB_ARDUINOHAL_PIN_STATUS_CAST)
+  #define RADIOLIB_ARDUINOHAL_PIN_STATUS_CAST
+  #endif
+  #if !defined(RADIOLIB_ARDUINOHAL_INTERRUPT_MODE_CAST)
+  #define RADIOLIB_ARDUINOHAL_INTERRUPT_MODE_CAST
+  #endif
```

Comment:
(Not important again) I wanted to know if it's not too "ugly" the repetition of `ifndef .. define .. endif`. It seems to be quite syntactically heavy but I haven't found an alternative.

Reply:
I would prefer it like this, if the user decides to define the macro in some other way (for example by calling gcc with -DRADIOLIB_ARDUINOHAL_INTERRUPT_MODE_CAST=something).

Reply:
Yes sure, I was rather searching a different syntax or way to do that.

Reply:
As far as I know there's nothing else, I've seen this syntax quite a lot.